components/style/properties/declaration_block.rs

```diff
@@ -935,13 +935,23 @@ impl<'a, 'b, 'i> AtRuleParser<'i> for PropertyDeclarationParser<'a, 'b> {
     type Error = SelectorParseError<'i, StyleParseError<'i>>;
 }
 
+fn is_non_mozilla_vendor_identifier(name: &str) -> bool {
+    (name.starts_with("-") && !name.starts_with("-moz-")) ||
+        name.starts_with("_")
+}
+
 impl<'a, 'b, 'i> DeclarationParser<'i> for PropertyDeclarationParser<'a, 'b> {
     type Declaration = Importance;
     type Error = SelectorParseError<'i, StyleParseError<'i>>;
 
     fn parse_value<'t>(&mut self, name: CompactCowStr<'i>, input: &mut Parser<'i, 't>)
                        -> Result<Importance, ParseError<'i>> {
-        let id = PropertyId::parse(name)?;
+        let id = match PropertyId::parse(&name) {
+            Ok(id) => id,
+            Err(()) if is_non_mozilla_vendor_identifier(&name) =>
```

Comment:
nit: I think this would be slightly easier to read like this:

```rust
    Err(()) => {
        return Err(if is_non_mozilla_vendor_identifier(&name) {
            PropertyDeclarationParseError::UnknownVendorProperty
        } else {
            // ..
        })
    }
```